app/src/main/java/com/github/campus_capture/bootcamp/fragments/MapsFragment.java

```diff
@@ -109,37 +130,98 @@ public class MapsFragment extends Fragment{
         int leftMargin = Math.round((5 * this.getContext().getResources().getDisplayMetrics().density));
         rlp.setMargins(leftMargin, 0, 0, bottomMargin);
 
-        //Listener when user clicks on the "my position" button
-        map.setOnMyLocationButtonClickListener(() -> {
-            Toast.makeText(getActivity(), "MyLocation button clicked", Toast.LENGTH_SHORT).show();
-            return false;
-        });
-
-        //Listener when user clicks on the "my position" blue dot
-        map.setOnMyLocationClickListener(location -> Toast.makeText(getActivity(), "Current location:\n" + location, Toast.LENGTH_LONG).show());
+        enableMyLocation();
 
-        // Add a marker at Satellite and move the camera
+        // Move the camera to the campus
         LatLng epfl = new LatLng(46.520536, 6.568318);
-        LatLng satellite = new LatLng(46.520544, 6.567825);
-        map.addMarker(new MarkerOptions().position(satellite).title("Satellite").snippet("5 ⭐"));
-        map.moveCamera(CameraUpdateFactory.newLatLngZoom(epfl, 15));
+        map.moveCamera(CameraUpdateFactory.newLatLngZoom(epfl, 17));
 
-        map.setOnMarkerClickListener(marker -> {
-            return false;
-        });
+        map.setInfoWindowAdapter(new LabelInfoWindowAdapter(getContext()));
+
+        zoneLabels = new HashMap<>();
 
         for (Zone zone : zoneDAO.getAll()){
             Polygon poly = map.addPolygon(new PolygonOptions().addAll(zone.getVertices()));
             poly.setStrokeWidth(0);
             poly.setFillColor(Color.argb(25, 255, 0, 0));
+            poly.setClickable(true);
+
+            zoneLabels.put(
+                    poly,
+                    map.addMarker(new MarkerOptions()
+                            .position(zone.getCenter())
+                            .icon(createPureTextIcon(zone.getName()))
+                            .title(zone.getName())
+                    ));
         }
 
         map.setOnPolygonClickListener(polygon ->{
+            Marker m = zoneLabels.get(polygon);
```

Comment:
This will maybe be a problem for later, since at every click on the zone, the informations are fetched from the DB, there will be a lot of traffic.